pkg/policy/api/l4.go

```diff
@@ -55,6 +55,59 @@ func (p PortProtocol) Covers(other PortProtocol) bool {
 	return true
 }
 
+// Secret is a reference to a secret, backed by k8s or local file system.
+type Secret struct {
+	// Namespace is the namespace in which the secret exists.
+	// Context of use determines the default value if left out (e.g., "default")
+	//
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
+
+	// Name is the name of the secret.
+	Name string `json:"name"`
+}
+
+// Equal returns true if 'a' and 'b' have the same contents.
+func (a *Secret) Equal(b *Secret) bool {
+	return a == nil && b == nil || a != nil && b != nil && *a == *b
+}
+
+// TLSContext provides TLS configuration via reference to either k8s secrets
+// or via filepath. If both are set, directory is given priority over
+// k8sSecrets.
+type TLSContext struct {
+	// Secret is the secret that contains the certificates and private key for the TLS context.
+	// By default, Cilium will search in this secret for the following items:
+	//  - 'ca.crt' - Which represents the trusted CA to verify remote source.
+	//  - 'tls.crt' - Which represents the public key certificate.
+	//  - 'tls.key' - Which represents the private key matching the public key certificate.
+	Secret *Secret `json:"secret,omitempty"`
+
+	// TrustedCA is the file name or k8s secret item name for the trusted CA.
+	// If omitted, 'ca.crt' is assumed, if it exists.
+	// If given, the item must exist.
+	TrustedCA string `json:"trustedCA,omitempty"`
+
+	// Certificate is the file name or k8s secret item name for the certificate chain.
+	// If omitted, 'tls.crt' is assumed, if it exists.
+	// If given, the item must exist.
+	Certificate string `json:"certificate,omitempty"`
+
+	// PrivateKey is the file name or k8s secret item name for the private key matching the certificate chain.
+	// If omitted, 'tls.key' is assumed, if it exists.
+	// If given, the item must exist.
+	PrivateKey string `json:"privateKey,omitempty"`
+}
+
+// Equal returns true if 'a' and 'b' have the same contents.
+func (a *TLSContext) Equal(b *TLSContext) bool {
```

Comment:
Same here.

Reply:
To avoid copying the structure if would be passed by value?

Reply:
Upon examining the call sites I recalled that the entities being compared are already pointers, so it is best to keep this `Equal()` function to avoid code repetition and keep the call sites more readable.